src/point_one/fusion_engine/parsers/fusion_engine_framer.cc

```diff
@@ -189,14 +200,24 @@ p1_ssize_t FusionEngineFramer::OnByte(bool quiet) {
       // Compute the full message size. If the message is too large to fit in
       // the buffer, we cannot parse it. Otherwise, start collecting the
       // message payload.
+      //
+      // Note that while we compute the current_message_size_ here, we
```

Comment:
ok i misunderstood this long comment before. what we're checking for is the case that the header is basically garbage and it's message length is causing a out of bounds condition.